packages/tracing/src/integrations/postgres.ts

```diff
@@ -0,0 +1,81 @@
+import { Hub } from '@sentry/hub';
+import { EventProcessor, Integration } from '@sentry/types';
+import { fill, logger } from '@sentry/utils';
+
+interface PgClient {
+  prototype: {
+    query: () => void | Promise<unknown>;
+  };
+}
+
+interface PostgresOptions {
+  client?: PgClient;
+}
+
+/** Tracing integration for node-postgres package */
+export class Postgres implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Postgres';
+
+  /**
+   * @inheritDoc
+   */
+  public name: string = Postgres.id;
+
+  private _client?: PgClient;
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options: PostgresOptions = {}) {
+    this._client = options.client;
+  }
+
+  /**
+   * @inheritDoc
+   */
+  public setupOnce(_: (callback: EventProcessor) => void, getCurrentHub: () => Hub): void {
+    if (!this._client) {
+      logger.error('PostgresIntegration is missing a Postgres.Client constructor');
+      return;
+    }
+
+    /**
+     * function (query, callback) => void
+     * function (query, params, callback) => void
+     * function (query) => Promise
+     * function (query, params) => Promise
+     */
+    fill(this._client.prototype, 'query', function(orig: () => void | Promise<unknown>) {
+      return function(this: unknown, config: unknown, values: unknown, callback: unknown) {
```

Comment:
The function we're returning is going to take the place of the original, right? And that original can take one of the four forms listed in the docstring? If so, then can we use the same variable names as above (or ones which show what we might have)?

```suggestion
      return function(this: unknown, queryStringOrObject: unknown, paramsOrCallback: unknown, callback?: unknown) {
```

Reply:
I did it this way (in all drivers) to use 1:1 naming from the original package.

Reply:
Ah, I see. That does make a certain amount of sense, but ugh - dealing with the overloading when they've picked particular (only sometimes accurate) names for the params. Gross.